### PR TITLE
Adding support for (tiny)DTLS

### PR DESCRIPTION
**Pull request by rfuentess**
This PR gives support for tinyDTLS (only PSK) by means of an updated to the aiocoap module. The user can enable this support with a new argument (`--dtls`).

The official repository implementation of aiocoap does not support tinyDTLS. Although there was an [effort on that topic](https://github.com/chrysn/aiocoap/issues/67#issuecomment-297917694). This forced me to use a [temporary fork](https://github.com/rfuentess/aiocoap/tree/tinydtls_server_v1) for supporting a server with DTLS, which is the reason of the use of r`equirements.txt` file. I hope that this specific commit can be suppressed in a future.

### CODE REVIEW

**Comment by rfuentess** [comment]
As commented briefly  offline. The extra option for CoAP Secure ports is removed. 

I originally put it for applications where the CoAP  and CoAP Secure server were running at the same time. However, I think that approach makes this app more complex than should be, as you commented.  

If the changes are OK then I'll proceed to squash them.

File: otaserver/app.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 from .server import OTAServerApplication
 from .coap import COAP_PORT
+from .coap import COAPS_PORT
```
Comment (aabadie) [inline otaserver/app.py]: This can be merged with the line above:
```python
from .coap import COAP_PORT, COAPS_PORT
```

File: otaserver/app.py [hunk 2]
```diff
@@ -40,6 +42,10 @@ def run(arguments=[]):
 
     parse_command_line()
 
+    # Custom port for CoAP overrides the one for CoAPS IF set
```
Comment (aabadie) [inline otaserver/app.py]: This comment is hard to understand. I would rephrase it like this:
Use CoAPs default port except if a custom CoAP port is set.

File: otaserver/app.py [hunk 3]
```diff
@@ -40,6 +42,10 @@ def run(arguments=[]):
...
+    if (options.dtls) and (options.coap_port is COAP_PORT):
```
Comment (aabadie) [inline otaserver/app.py]: parenthesis are not needed.

File: otaserver/coap.py [hunk 4]
```diff
@@ -13,7 +13,7 @@
 
 
 COAP_PORT = 5683
-
```
Comment (aabadie) [inline otaserver/coap.py]: 2 empty lines are needed (otherwise PEP8 error)

File: otaserver/coap.py [hunk 5]
```diff
@@ -91,14 +91,15 @@ def render_get(self, request):
 class CoapController():
     """CoAP controller with CoAP server inside."""
 
-    def __init__(self, fw_path, port=COAP_PORT):
+    def __init__(self, fw_path, port=COAP_PORT, dtls_enabled=False):
```
Comment (aabadie) [inline otaserver/coap.py]: I would rename `dtls_enabled` to `use_dtls` or `use_coaps`. What do you think ?

File: otaserver/app.py [hunk 6]
```diff
@@ -28,7 +29,8 @@ def parse_command_line():
            default=__UPLOAD_PATH__,
            help="Path where uploaded files are stored.")
     define("port", default=8080, help="Web application HTTP port.")
-    define("coap_port", default=COAP_PORT, help="CoAP server port.")
+    define("coap_port", default=COAP_PORT, help="CoAP(S) server port.")
+    define("dtls", default=False, help="Enable CoAP Secure.")
```
Comment (aabadie) [inline otaserver/app.py]: I would rename this option to `use-dtls` or `use-coaps`. What do you think ?

Comment (rfuentess) [reply otaserver/app.py]: Yes, I think that for this application running two services in two different ports, having a more clearly name for all the options related to CoAPS is better.

File: otaserver/coap.py (hunk 5 above)
Comment (rfuentess) [reply otaserver/coap.py]: For this one, I'm not inclined to do it. This is mostly because the way aiocoap is handling the CoAP(s) packets, and the use of names in that module.

Comment (aabadie) [reply otaserver/coap.py]: With your last changes, I tend to agree with you. Let's keep this one.

**Comment by aabadie** [comment]
General comment: do you think your work on aiocoap has any chance of being merged one day ? Why not opening a PR on aiocoap ?

**Comment by rfuentess** [comment]
Yes, I contacted the author of aiocoap. He is retaking the work with DTLS. I hope to be able to speak with him again on the RIOT Summit to discuss a little more of the pending issues for having a better support of tinydtls on python.

After the summit, I'll work again on my aicoap's branch and make the PR.

**Comment by aabadie** [comment]
> I hope to be able to speak with him again on the RIOT Summit

Will he attend the summit ? That's awesome !

**Comment by rfuentess** [comment]
Just a fast note, current work in aiocoap rewrites too much of their original beta implementation of tinydtls. Also, it's seems that is not ideal when a client and server are running at the same time, as I discovered it today.

Therefore, after the RIOT summit I will rebase the master  to try to use their proposed model, once more time. And of course, the PR for the official repository.
